src/ert/scheduler/lsf_driver.py

```diff
@@ -236,15 +273,26 @@ async def poll(self) -> None:
                 logger.warning(
                     f"bjobs gave returncode {process.returncode} and error {stderr.decode()}"
                 )
-            stat = _Stat(**parse_bjobs(stdout.decode(errors="ignore")))
-            for job_id, job in stat.jobs.items():
-                await self._process_job_update(job_id, job)
-            missing_in_bjobs_output = set(self._jobs) - set(stat.jobs.keys())
+            bjobs_states = _Stat(**parse_bjobs(stdout.decode(errors="ignore")))
+            missing_in_bjobs_output = set(self._jobs) - set(bjobs_states.jobs.keys())
             if missing_in_bjobs_output:
-                logger.warning(
-                    f"bjobs did not give status for job_ids {missing_in_bjobs_output}"
+                logger.debug(f"bhist is needed for job ids: {missing_in_bjobs_output}")
+
+            bhist_states = await self._poll_once_by_bhist(missing_in_bjobs_output)
```

Comment:
hm, can the if statement be moved out from the function?:
```python
if not missing_job_ids:
```
Since the function title is misleading then.

Reply:
Added a commit now that tries this, it might look slightly better.